promql/engine.go

```diff
@@ -2937,6 +2941,10 @@ func (ev *evaluator) aggregation(e *parser.AggregateExpr, q float64, inputMatrix
 		groups[i].seen = false
 	}
 
+	if e.Op == parser.QUANTILE && (math.IsNaN(q) || q < 0 || q > 1) {
+		annos.Add(annotations.NewInvalidQuantileWarning(q, e.Param.PositionRange()))
```

Comment:
Since this annotation is not specific to the affected series, we need at at most once. I think we can do this once ahead of the iteration by checking the max and min and if any is NaN. (See my idea about the interface above.)

If we have many steps, adding all those annotations, that then are promptly removed as duplicates again, could be quite some churn.

Reply:
Done!